metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -349,25 +351,104 @@ def _process_view(
+            if view_definition is None:
+                view_definition = ""
+            else:
+                view_definition = str(view_definition)
+        except NotImplementedError:
+            view_definition = ""
+        properties["view_definition"] = view_definition
+        properties["is_view"] = "True"
+        dataset_urn = make_dataset_urn_with_platform_instance(
+            self.platform,
+            dataset_name,
+            self.config.platform_instance,
+            self.config.env,
+        )
+        dataset_snapshot = DatasetSnapshot(
+            urn=dataset_urn,
+            aspects=[StatusClass(removed=False)],
+        )
+        db_name = self.get_db_name(inspector)
+        yield from self.add_table_to_schema_container(
+            dataset_urn=dataset_urn,
+            db_name=db_name,
+            schema=schema,
+        )
+
+        dataset_properties = DatasetPropertiesClass(
+            name=view,
+            description=description,
+            customProperties=properties,
+        )
+        dataset_snapshot.aspects.append(dataset_properties)
+        if schema_metadata:
+            dataset_snapshot.aspects.append(schema_metadata)
+        mce = MetadataChangeEvent(proposedSnapshot=dataset_snapshot)
+        yield SqlWorkUnit(id=dataset_name, mce=mce)
+        dpi_aspect = self.get_dataplatform_instance_aspect(dataset_urn=dataset_urn)
+        if dpi_aspect:
+            yield dpi_aspect
+        yield MetadataChangeProposalWrapper(
+            entityUrn=dataset_urn,
+            aspect=SubTypesClass(typeNames=[DatasetSubTypes.VIEW]),
+        ).as_workunit()
+        if "view_definition" in properties:
+            view_definition_string = properties["view_definition"]
+            view_properties_aspect = ViewPropertiesClass(
+                materialized=False, viewLanguage="SQL", viewLogic=view_definition_string
+            )
+            yield MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn,
+                aspect=view_properties_aspect,
+            ).as_workunit()
+
+        if self.config.domain and self.domain_registry:
+            yield from get_domain_wu(
+                dataset_name=dataset_name,
+                entity_urn=dataset_urn,
+                domain_config=sql_config.domain,
+                domain_registry=self.domain_registry,
+            )
+
         dataset_urn = make_dataset_urn_with_platform_instance(
             self.platform,
             dataset_name,
             self.config.platform_instance,
             self.config.env,
         )
-        view_owner = self._get_owner_information(view, "view")
+
+        view_owner = inspector.get_view_owner(view, schema)
         yield from add_owner_to_entity_wu(
             entity_type="dataset",
             entity_urn=dataset_urn,
             owner_urn=f"urn:li:corpuser:{view_owner}",
         )
-        yield from super()._process_view(
```

Comment:
why can we not call super()._process_view here?

a lot of this logic looks copy-pasted

Reply:
we had to use copy-pasted code due to the columns variable, we cant use get_columns as it fetched queries based on columns. we had to create a new function for columns and properties. and since it is not possible to override the parent class variable data. we were not able to use super here and due to this all copy-pasted code.